packages/asset-swapper/src/utils/market_operation_utils/types.ts

```diff
@@ -268,19 +268,29 @@ export interface HopInfo {
     returnData: string;
 }
 
+export interface UniswapV3PathAmount {
+    uniswapPath: string;
+    inputAmount: BigNumber;
+    gasUsed: number;
+}
 export interface UniswapV3FillData extends FillData {
     tokenAddressPath: string[];
     router: string;
-    pathAmounts: Array<{ uniswapPath: string; inputAmount: BigNumber }>;
+    pathAmounts: UniswapV3PathAmount[];
 }
 
 export interface KyberDmmFillData extends UniswapV2FillData {
     poolsPath: string[];
 }
 
-export interface FinalUniswapV3FillData extends Omit<UniswapV3FillData, 'uniswapPaths'> {
+export const isFinalUniswapV3FillData = (
+    data: UniswapV3FillData | FinalUniswapV3FillData,
+): data is FinalUniswapV3FillData => !!(data as FinalUniswapV3FillData).uniswapPath;
```

Comment:
no closures being used so why not a regular function?

Reply:
No reason other than I like the shorter `=>` version 😂  I don't believe there is any performance difference

Reply:
Don't think there is in this case but we lean towards normal/named functions where possible as a rule because they show up in stack traces.

Reply:
Ah, that makes sense! Ty 🙏